ui/src/message_popup/author/question_page.jsx

```diff
@@ -25,21 +27,30 @@ export default React.createClass({
 
   propTypes: {
     originalQuestion: React.PropTypes.object,
+    allQuestions: React.PropTypes.object
```

Comment:
Could we remove this altogether from `QuestionPage`?  It looks like the only place it's used is to figure out the next `questionId` when creating a new one.  I think it will simplify to leave the actual saving as the responsibility of a container outside of `question_page`, and have it handle making ids and communicating with the database when it's time to save, edit or delete.

So concretely, this would mean refactoring methods here like `onSaveEdit` so that they all ultimately call something like `onQuestionsChanged`.  This will reduce some of the duplication between then now.  We can then factor the `onQuestionChanged` logic out of this component so that `QuestionPage` stays focused on only dealing with a single question.  The `onQuestionsChanged` logic can then be pulled out into the top-level `new_question_page` or `edit_question_page` components.

Let me know if I'm explaining this well or if it's easier to talk out in person!